drivers/sensor/ina219/ina219.c

```diff
@@ -0,0 +1,353 @@
+/*
+ * Copyright (c) 2021 Leonard Pollak
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT ti_ina219
+
+#include <device.h>
+#include <drivers/i2c.h>
+#include <kernel.h>
+#include <drivers/sensor.h>
+#include <sys/__assert.h>
+#include <logging/log.h>
+
+#include <drivers/sensor/ina219.h>
+#include "ina219.h"
+
+LOG_MODULE_REGISTER(INA219, CONFIG_SENSOR_LOG_LEVEL);
+
+static int ina219_reg_read(const struct device *dev,
+		uint8_t reg_addr,
+		uint16_t *reg_data)
+{
+	const struct ina219_config *cfg = dev->config;
+	uint8_t rx_buf[2];
+	int rc;
+
+	rc = i2c_write_read(cfg->bus, cfg->i2c_addr,
+			&reg_addr, sizeof(reg_addr),
+			rx_buf, sizeof(rx_buf));
+
+	*reg_data = ((uint16_t)rx_buf[0] << 8 | rx_buf[1]);
```

Comment:
`byteorder.h` API should be used

Reply:
I have done this for both the read and write functions.